include/nuttx/net/offload.h

```diff
@@ -0,0 +1,474 @@
+ * Name: netdev_enable_gso
+ *
+ * Description:
+ *   Enable the GSO function for the dev.
+ *   Set the features and GSO-related maximum value.
+ *
+ * Input Parameters:
+ *    dev -  The structure of the network driver that enables the GSO
+ *           function.
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void netdev_enable_gso(FAR struct net_driver_s *dev);
+
+/****************************************************************************
+ * Name: devif_needs_gso
+ *
+ * Description:
+ *   Check whether the packet needs to be segmented.
+ *
+ * Input Parameters:
+ *    pkt  - the packet to be sent by the NIC driver.
+ *    features  - the features are supported by the target dev of the packet.
+ *
+ * Returned Value:
+ *    The false indicates that the packet does not to be divided.
+ *    The true indicates that the packet needs to be divided.
+ *
+ ****************************************************************************/
+
+bool devif_needs_gso(FAR struct iob_s *pkt, int features);
+
+/****************************************************************************
+ * Name: devif_gso_segment
+ *
+ * Description:
+ *   Divide the packet into multiple packets based on the features and
+ *   the GSO information of the packet.
+ *
+ * Input Parameters:
+ *   pkt  - the packet to be sent by the NIC driver.
+ *   features  - the features are supported by the target dev of the packet.
+ *
+ * Returned Value:
+ *   The NULL indicates that the devision is failure.
+ *   The segs is the head of the multiple packets.
+ *
+ ****************************************************************************/
+
+FAR struct iob_s *devif_gso_segment(FAR struct iob_s *pkt,
+                                    uint32_t features);
+
+/****************************************************************************
+ * Name: devif_gso_list_free
+ *
+ * Description:
+ *   Free the packet allocated by pkt_segment and
+ *   clear the GSO information of the fisrt packet.
+ *
+ * Input Parameters:
+ *    pkt  - The GSO packets includes the segments list.
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void devif_gso_list_free(FAR struct iob_s *pkt);
+
+/****************************************************************************
+ * Name: pkt_segment
+ *
+ * Description:
+ *   Segment the packet to be divided, and add the header information for all
+ *   the segments.
+ *
+ * Input Parameters:
+ *   pkt  - the packet needs to be divided.
+ *   features  - the features are supported by the target dev of the packet.
+ *
+ * Returned Value:
+ *   The NULL indicates that the devision is failure.
+ *   The segs is the head of the multiple packets.
+ *
+ ****************************************************************************/
+
+FAR struct iob_s *pkt_segment(FAR struct iob_s *pkt, uint32_t features);
```

Comment:
```suggestion
FAR struct iob_s *devif_pkt_segment(FAR struct iob_s *pkt, uint32_t features);
```
Unify all function naming

Reply:
Done